books/workshops/references/xdoc-generation.lisp

```diff
@@ -0,0 +1,159 @@
+  (cond ((equal entry-type "article") "Journal Article")
+        ((equal entry-type "inproceedings") "Conference Paper")
+        ((equal entry-type "incollection") "Book Chapter")
+        ((equal entry-type "techreport") "Technical Report")
+        (t (concatenate 'string
+                        (string-upcase (subseq entry-type 0 1))
+                        (subseq entry-type 1 nil)))))
+
+(defun generate-bibtex-entry-XDOC (entry)
+  "Generate XDOC list item for a single BibTeX entry"
+  (let* ((entry-type (get-entry-field "type" entry))
+         (fields (get-entry-field "fields" entry))
+         (author (get-field-value "author" fields))
+         (title (get-field-value "title" fields))
+         (year (get-field-value "year" fields))
+         (pages (get-field-value "pages" fields))
+         (publisher (get-field-value "publisher" fields))
+         (booktitle (get-field-value "booktitle" fields))
+         (month (get-field-value "month" fields))
+         (isbn (get-field-value "isbn" fields))
+         (doi (get-field-value "doi" fields)))
+
+    (concatenate 'string
+                 "<li><b>" (format-title title) "</b>, "
+                 (format-author-names author)
+                 (if booktitle
+                     (concatenate 'string ", in <i>" (strip-brackets booktitle) "</i>")
+                   "")
+                 (if publisher
+                     (concatenate 'string ", " (strip-brackets publisher))
+                   "")
+                 (if year
+                     (concatenate 'string ", " (strip-brackets year))
+                   "")
+                 (if month
+                     (concatenate 'string ", " (strip-brackets month))
+                   "")
+                 (if pages
+                     (concatenate 'string ", " (format-pages pages))
+                   "")
+                 (if isbn
+                     (concatenate 'string ", ISBN: " (strip-brackets isbn))
+                   "")
+                 (if doi
+                     (concatenate 'string ", DOI: " (strip-brackets doi))
+                   "")
+                 ". [" (format-entry-type entry-type) "]"
+                 "</li>")))
+
+(defun generate-bibtex-entries-XDOC (entries)
+  "Generate XDOC list items for all BibTeX entries"
+  (if (endp entries)
+      ""
+    (let ((entry-pair (car entries)))
+      (if (and (consp entry-pair)
+               (stringp (car entry-pair))
+               (alistp (cdr entry-pair)))
+          (concatenate 'string
+                       (generate-bibtex-entry-XDOC (cdr entry-pair))
+                       (generate-bibtex-entries-XDOC (cdr entries)))
+        (generate-bibtex-entries-XDOC (cdr entries))))))
+
+
+(defun generate-workshop-documentation-xdoc (entries)
+  "Generate single XDOC documentation for all workshop entries"
+  (let ((entry-count (len entries))
+        (entries (generate-bibtex-entries-XDOC entries)))
+    `(defxdoc acl2-workshops
+       :parents (pubs-papers)
```

Comment:
This should actually be `pubs::pubs-papers` because the `pubs-papers` topic is in the `pubs` package. After you make this change, cert.pl will need help certifying the book because it will need to know about the `pubs` package. You'll need to create a `xdoc-generation.acl2` file, which `cert.pl` will automatically run before certifying `xdoc-generation.lisp`. The contents of the file just need to be `(ld "../../doc/publications.acl2")`. This will load the file where the `pubs` package is defined.